gdb-7.8/gdb/breakpoint.c

```diff
@@ -13614,6 +13617,40 @@ bkpt_resources_needed (const struct bp_location *bl)
 }
 
 static enum print_stop_action
+gpu_bkpt_trigger_print_it (bpstat bs)
+{
+  struct breakpoint *b;
+  const struct bp_location *bl;
+  int bp_temp;
+  struct ui_out *uiout = current_uiout;
+
+  gdb_assert (bs->bp_location_at != NULL);
+
+  bl = bs->bp_location_at;
+  b = bs->breakpoint_at;
+
+  bp_temp = b->disposition == disp_del;
+  if (bl->address != bl->requested_address)
+    breakpoint_adjustment_warning (bl->requested_address,
+				   bl->address,
+				   b->number, 1);
+  annotate_breakpoint (b->number);
+  if (bp_temp)
+    ui_out_text (uiout, "\nTemporary breakpoint triggered by GPU Kernel Breakpoint");
+  else
+    ui_out_text (uiout, "\nStopped on GPU Breakpoint");
+  if (ui_out_is_mi_like_p (uiout))
+    {
+      ui_out_field_string (uiout, "reason",
+			   async_reason_lookup (EXEC_ASYNC_BREAKPOINT_HIT));
+      ui_out_field_string (uiout, "disp", bpdisp_text (b->disposition));
+    }
+  ui_out_text (uiout, ", ");
+
+  return PRINT_SRC_AND_LOC;
```

Comment:
I was trying this out internally and I noticed that it should be 
"return PRINT_NOTHING". 
The PRINT_SRC_AND_LOC prints information about the trigger function. I feel that the trigger is something internal to the debugger and not related to  debugging the application.